### ISSUE TITLE
Mempool Offline / Reconnecting

### ISSUE THREAD

**Issue by MyNameIsOka** [mynodebtc/mynode#668]
**Describe the bug**
Since the update to 0.2.57 and the update to Mempool v2.4.0, Mempool can't connect to the node anymore.
It seems that there is an error in the mempool backend trying to access the database. 
It says:
```
...
wait-for-it.sh: waiting 720 seconds for db:3306
wait-for-it.sh: db:3306 is available after 0 seconds
git not found, cannot parse git hash
Jun 22 00:26:02 [81] NOTICE: Starting Mempool Server... (?)
Jun 22 00:26:02 [81] INFO: Database connection established.
Jun 22 00:26:02 [81] INFO: Connected to Electrum Server at 172.26.0.1:50001 (["electrs/0.9.3","1.4"])
Jun 22 00:26:02 [81] ERR: MIGRATIONS: Unable to create required tables, aborting in 10 seconds. Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json NOT NULL,
      median_fee double unsigned NOT NULL,
      PRIMARY KEY (hei' at line 12
wait-for-it.sh: waiting 720 seconds for db:3306
wait-for-it.sh: db:3306 is available after 0 seconds
git not found, cannot parse git hash
Jun 22 00:26:15 [82] NOTICE: Starting Mempool Server... (?)
Jun 22 00:26:15 [82] INFO: Database connection established.
Jun 22 00:26:15 [82] INFO: Connected to Electrum Server at 172.26.0.1:50001 (["electrs/0.9.3","1.4"])
Jun 22 00:26:15 [82] ERR: MIGRATIONS: Unable to create required tables, aborting in 10 seconds. Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json NOT NULL,
      median_fee double unsigned NOT NULL,
      PRIMARY KEY (hei' at line 12
wait-for-it.sh: waiting 720 seconds for db:3306
wait-for-it.sh: db:3306 is available after 0 seconds
git not found, cannot parse git hash
Jun 22 00:26:35 [82] NOTICE: Starting Mempool Server... (?)
Jun 22 00:26:36 [82] INFO: Database connection established.
Jun 22 00:26:36 [82] INFO: Connected to Electrum Server at 172.26.0.1:50001 (["electrs/0.9.3","1.4"])
Jun 22 00:26:36 [82] ERR: MIGRATIONS: Unable to create required tables, aborting in 10 seconds. Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json NOT NULL,
      median_fee double unsigned NOT NULL,
      PRIMARY KEY (hei' at line 12
...
```
**Expected behavior**
Should connect to the node.

**Screenshots**
Mempool screenshot:
![image](https://user-images.githubusercontent.com/18796117/174918411-68d91175-cd2e-4219-8887-6e2ced984c06.png)

Screenshot from Docker container:
![image](https://user-images.githubusercontent.com/18796117/174918496-aea1a0b3-b367-40b4-83b9-575a7cfaf6ef.png)

**Desktop (please complete the following information):**
- Every browser
- OSX 12.4

**MyNode hardware (please complete the following information):**
 - Device: Raspi4
 - MyNodeBTC version: 0.2.57

**Comment by tehelsper** [comment]
That looks like an upgrade issue. Can you try re-installing mempool?

**Comment by MyNameIsOka** [comment]
Thank you for your response. I already tried that but it didn't help.

**Comment by tehelsper** [comment]
Are you on 32-bit or 64-bit Raspi 4? I would re-install again and save the Docker Images log on the status page to see if there are any setup errors. Each re-install should re-init the database.

**Comment by MyNameIsOka** [comment]
I'm on 32-bit. 
Okay, I will try that.

**Comment by MyNameIsOka** [comment]
Here are the logs right after reinstallation but to be honest, I don't see any error except 
- the database migrations error and
- the git hash error

```
Jun 21 23:28:15 myNode mempool[8700]: api_1  | wait-for-it.sh: db:3306 is available after 0 seconds
Jun 21 23:28:15 myNode mempool[8700]: api_1  | wait-for-it.sh: waiting 720 seconds for db:3306
Jun 21 23:28:02 myNode mempool[8700]: api_1  |       PRIMARY KEY (hei' at line 12
Jun 21 23:28:02 myNode mempool[8700]: api_1  |       median_fee double unsigned NOT NULL,
Jun 21 23:28:02 myNode mempool[8700]: api_1  | Jun 22 04:28:02 [92] ERR: MIGRATIONS: Unable to create required tables, aborting in 10 seconds. Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json NOT NULL,
Jun 21 23:28:02 myNode systemd[1]: Started Mempool.
Jun 21 23:28:02 myNode mempool[8701]: + exit 0
Jun 21 23:28:02 myNode mempool[8701]: performance_schema == *\m\e\m\p\o\o\l* ]]
Jun 21 23:28:02 myNode mempool[8701]: mysql
Jun 21 23:28:02 myNode mempool[8701]: mempool
Jun 21 23:28:02 myNode mempool[8701]: information_schema
Jun 21 23:28:02 myNode mempool[8701]: + [[ Database
Jun 21 23:28:02 myNode mempool[8701]: performance_schema == *\i\n\f\o\r\m\a\t\i\o\n\_\s\c\h\e\m\a* ]]
Jun 21 23:28:02 myNode mempool[8701]: mysql
Jun 21 23:28:02 myNode mempool[8701]: mempool
Jun 21 23:28:02 myNode mempool[8701]: information_schema
Jun 21 23:28:02 myNode mempool[8701]: + [[ Database
Jun 21 23:28:02 myNode mempool[8701]: performance_schema'
Jun 21 23:28:02 myNode mempool[8701]: mysql
Jun 21 23:28:02 myNode mempool[8701]: mempool
Jun 21 23:28:02 myNode mempool[8701]: information_schema
Jun 21 23:28:02 myNode mempool[8701]: + databases='Database
Jun 21 23:28:00 myNode mempool[8701]: ++ docker exec -i mempool_db_1 mysql -uroot -padmin -e 'SHOW DATABASES;'
Jun 21 23:28:00 myNode mempool[8700]: api_1  | Jun 22 04:28:00 [92] INFO: Database connection established.
Jun 21 23:28:00 myNode mempool[8700]: api_1  | Jun 22 04:28:00 [92] INFO: Connected to Electrum Server at 172.26.0.1:50001 (["electrs/0.9.3","1.4"])
Jun 21 23:28:00 myNode mempool[8700]: api_1  | Jun 22 04:28:00 [92] NOTICE: Starting Mempool Server... (?)
Jun 21 23:27:59 myNode mempool[8700]: api_1  | git not found, cannot parse git hash
Jun 21 23:27:58 myNode mempool[8700]: api_1  | wait-for-it.sh: db:3306 is available after 5 seconds
Jun 21 23:27:57 myNode mempool[8700]: db_1   | Version: '5.5.60-0+deb7u1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  (Debian)
Jun 21 23:27:57 myNode mempool[8700]: db_1   | 220622  4:27:57 [Note] mysqld: ready for connections.
Jun 21 23:27:57 myNode mempool[8700]: db_1   | 220622  4:27:57 [Note] Event Scheduler: Loaded 0 events
Jun 21 23:27:57 myNode mempool[8700]: db_1   | 220622  4:27:57 [Note] Server socket created on IP: '0.0.0.0'.
Jun 21 23:27:57 myNode mempool[8700]: db_1   | 220622  4:27:57 [Note]   - '0.0.0.0' resolves to '0.0.0.0';
Jun 21 23:27:57 myNode mempool[8700]: db_1   | 220622  4:27:57 [Note] Server hostname (bind-address): '0.0.0.0'; port: 3306
Jun 21 23:27:57 myNode mempool[8700]: db_1   | 220622  4:27:57 InnoDB: 5.5.60 started; log sequence number 0
Jun 21 23:27:56 myNode mempool[8700]: db_1   | 220622  4:27:56  InnoDB: Waiting for the background threads to start
Jun 21 23:27:56 myNode mempool[8700]: db_1   | InnoDB: Foreign key constraint system tables created
Jun 21 23:27:56 myNode mempool[8700]: db_1   | InnoDB: Creating foreign key constraint system tables
Jun 21 23:27:56 myNode mempool[8700]: db_1   | InnoDB: 127 rollback segment(s) active.
Jun 21 23:27:55 myNode mempool[8701]: + sleep 5s
Jun 21 23:27:55 myNode mempool[8701]: + '[' true == true ']'
Jun 21 23:27:55 myNode mempool[8701]: + isRunning=true
Jun 21 23:27:55 myNode mempool[8700]: db_1   | InnoDB: Doublewrite buffer created
Jun 21 23:27:55 myNode mempool[8701]: ++ docker inspect '--format={{.State.Running}}' mempool_db_1
Jun 21 23:27:55 myNode mempool[8701]: + '[' 1 ']'
Jun 21 23:27:55 myNode mempool[8701]: + isRunning=
Jun 21 23:27:55 myNode mempool[8700]: db_1   | InnoDB: Doublewrite buffer not found: creating new
Jun 21 23:27:55 myNode mempool[8700]: db_1   | InnoDB: Database physically writes the file full: wait...
Jun 21 23:27:55 myNode mempool[8700]: db_1   | InnoDB: Setting log file ./ib_logfile1 size to 5 MB
Jun 21 23:27:55 myNode mempool[8700]: db_1   | 220622  4:27:55  InnoDB: Log file ./ib_logfile1 did not exist: new to be created
Jun 21 23:27:55 myNode mempool[8700]: db_1   | InnoDB: Database physically writes the file full: wait...
Jun 21 23:27:55 myNode mempool[8700]: db_1   | InnoDB: Setting log file ./ib_logfile0 size to 5 MB
Jun 21 23:27:55 myNode mempool[8700]: db_1   | 220622  4:27:55  InnoDB: Log file ./ib_logfile0 did not exist: new to be created
Jun 21 23:27:54 myNode mempool[8700]: db_1   | InnoDB: Database physically writes the file full: wait...
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54  InnoDB: Setting file ./ibdata1 size to 10 MB
Jun 21 23:27:54 myNode mempool[8700]: db_1   | InnoDB: a new database to be created!
Jun 21 23:27:54 myNode mempool[8700]: db_1   | InnoDB: The first specified data file ./ibdata1 did not exist:
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 InnoDB: Completed initialization of buffer pool
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 InnoDB: Initializing buffer pool, size = 128.0M
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 InnoDB: Using Linux native AIO
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 InnoDB: Compressed tables use zlib 1.2.7
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 InnoDB: Mutexes and rw_locks use GCC atomic builtins
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 InnoDB: The InnoDB memory heap is disabled
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Note] Plugin 'FEDERATED' is disabled.
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Warning] Using unique option prefix myisam-recover instead of myisam-recover-options is deprecated and will be removed in a future release. Please use the full name instead.
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Note] mysqld (mysqld 5.5.60-0+deb7u1) starting as process 1 ...
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Warning] Using unique option prefix key_buffer instead of key_buffer_size is deprecated and will be removed in a future release. Please use the full name instead.
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | /entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
Jun 21 23:27:54 myNode mempool[8700]: db_1   | Database initialized
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | Please report any problems at http://bugs.mysql.com/
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | cd /usr/mysql-test ; perl mysql-test-run.pl
Jun 21 23:27:54 myNode mempool[8700]: db_1   | You can test the MySQL daemon with mysql-test-run.pl
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | cd /usr ; /usr/bin/mysqld_safe &
Jun 21 23:27:54 myNode mempool[8700]: db_1   | You can start the MySQL daemon with:
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | See the manual for more instructions.
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | strongly recommended for production servers.
Jun 21 23:27:54 myNode mempool[8700]: db_1   | databases and anonymous user created by default.  This is
Jun 21 23:27:54 myNode mempool[8700]: db_1   | which will also give you the option of removing the test
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | /usr/bin/mysql_secure_installation
Jun 21 23:27:54 myNode mempool[8700]: db_1   | Alternatively you can run:
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | /usr/bin/mysqladmin -u root -h 2932f54eb469 password 'new-password'
Jun 21 23:27:54 myNode mempool[8700]: db_1   | /usr/bin/mysqladmin -u root password 'new-password'
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | To do so, start the server, then issue the following commands:
Jun 21 23:27:54 myNode mempool[8700]: db_1   | PLEASE REMEMBER TO SET A PASSWORD FOR THE MySQL root USER !
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | support-files/mysql.server to the right place for your system
Jun 21 23:27:54 myNode mempool[8700]: db_1   | To start mysqld at boot time you have to copy
Jun 21 23:27:54 myNode mempool[8700]: db_1   |
Jun 21 23:27:54 myNode mempool[8700]: db_1   | OK
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Note] /usr/sbin/mysqld (mysqld 5.5.60-0+deb7u1) starting as process 47 ...
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Note] Ignoring --secure-file-priv value as server is running with --bootstrap.
Jun 21 23:27:54 myNode mempool[8700]: db_1   | 220622  4:27:54 [Warning] Using unique option prefix key_buffer instead of key_buffer_size is deprecated and will be removed in a future release. Please use the full name instead.
Jun 21 23:27:54 myNode mempool[8700]: db_1   | Filling help tables...
Jun 21 23:27:54 myNode mempool[8700]: db_1   | OK
Jun 21 23:27:53 myNode mempool[8700]: db_1   | 220622  4:27:53 [Note] /usr/sbin/mysqld (mysqld 5.5.60-0+deb7u1) starting as process 41 ...
Jun 21 23:27:53 myNode mempool[8700]: db_1   | 220622  4:27:53 [Note] Ignoring --secure-file-priv value as server is running with --bootstrap.
Jun 21 23:27:53 myNode mempool[8700]: db_1   | 220622  4:27:53 [Warning] Using unique option prefix key_buffer instead of key_buffer_size is deprecated and will be removed in a future release. Please use the full name instead.
Jun 21 23:27:53 myNode mempool[8700]: db_1   | Installing MySQL system tables...
Jun 21 23:27:53 myNode mempool[8700]: db_1   | Initializing database
Jun 21 23:27:53 myNode mempool[8700]: api_1  | wait-for-it.sh: waiting 720 seconds for db:3306
Jun 21 23:27:53 myNode mempool[8700]: Attaching to mempool_db_1, mempool_api_1, mempool_web_1
Jun 21 23:27:53 myNode mempool[8700]: Creating mempool_web_1 ... done
Jun 21 23:27:53 myNode mempool[8700]: Creating mempool_api_1 ... done
Jun 21 23:27:52 myNode mempool[8700]: Creating mempool_db_1  ... done
Jun 21 23:27:50 myNode mempool[8700]: Creating mempool_web_1 ...
Jun 21 23:27:49 myNode mempool[8700]: Creating mempool_db_1  ...
Jun 21 23:27:49 myNode mempool[8700]: Creating mempool_api_1 ...
Jun 21 23:27:49 myNode mempool[8700]: Creating network "mempool_default" with driver "bridge"
...
```

**Comment by MyNameIsOka** [comment]
Switching to 64-bit resolved the issue.
